code/modules/mob/living/simple_animal/hostile/ooze.dm

```diff
@@ -0,0 +1,490 @@
+
+	if(ooze.ooze_nutrition < 5)
+		to_chat(ooze, "<span class='warning'>You need at least 5 nutrition to launch a mending globule.</span>")
+		remove_ranged_ability()
+		return
+
+	ooze.visible_message("<span class='nicegreen>[ooze] launches a mending globule!</span>", "<span class='notice'>You launch a mending globule.</span>")
+	var/obj/projectile/globule/globule = new (ooze.loc)
+	globule.preparePixelProjectile(target, ooze, params)
+	globule.def_zone = ooze.zone_selected
+	globule.fire()
+	ooze.adjust_ooze_nutrition(-5)
+	remove_ranged_ability()
+	current_cooldown = world.time + cooldown
+
+	return TRUE
+
+/obj/effect/proc_holder/globules/on_lose(mob/living/carbon/user)
+	remove_ranged_ability()
+
+///This projectile embeds into mobs and heals them over time.
+/obj/projectile/globule
+	name = "mending globule"
+	icon_state = "glob_projectile"
+	shrapnel_type = /obj/item/mending_globule
+	embedding = list("embed_chance" = 100, ignore_throwspeed_threshold = TRUE, "pain_mult" = 0, "jostle_pain_mult" = 0, "fall chance" = 0.5)
+	nodamage = TRUE
+	damage = 0
+
+///This item is what is embedded into the mob, and actually handles healing of mending globules
+/obj/item/mending_globule
+	name = "mending globule"
+	desc = "It somehow heals those who touch it."
+	icon = 'icons/obj/xenobiology/vatgrowing.dmi'
+	icon_state = "globule"
+	embedding = list("embed_chance" = 100, ignore_throwspeed_threshold = TRUE, "pain_mult" = 0, "jostle_pain_mult" = 0, "fall chance" = 0.5)
+	var/obj/item/bodypart/bodypart
+	var/heals_left = 35
+
+/obj/item/mending_globule/embedded(mob/living/carbon/human/embedded_mob, obj/item/bodypart/part)
+	. = ..()
+	if(!istype(part))
+		return
+	bodypart = part
+	START_PROCESSING(SSobj, src)
+
+/obj/item/mending_globule/unembedded()
+	. = ..()
+	bodypart = null
+	STOP_PROCESSING(SSobj, src)
+
+///Handles the healing of the mending globule
+/obj/item/mending_globule/process()
+	if(!bodypart) //this is fucked
+		return FALSE
+	bodypart.heal_damage(1,1)
+	heals_left--
+	if(heals_left <= 0)
+		qdel(src)
+
+///This action lets you put a mob inside of a cacoon that will inject it with some chemicals.
+/datum/action/cooldown/gel_cocoon
+	name = "Gel Cocoon"
+	desc = "Puts a mob inside of a cocoon, allowing it to slowly heal."
+	background_icon_state = "bg_hive"
+	icon_icon = 'icons/mob/actions/actions_slime.dmi'
+	button_icon_state = "gel_cocoon"
+	check_flags = AB_CHECK_CONSCIOUS|AB_CHECK_STUN
+	cooldown_time = 10 SECONDS
+
+///Try to put the pulled mob in a cocoon
+/datum/action/cooldown/gel_cocoon/Trigger()
+	. = ..()
+	if(!.)
+		return
+	var/mob/living/simple_animal/hostile/ooze/grapes/ooze = owner
+	if(!ooze.pulling)
+		to_chat(src, "<span class='warning'>You need to be pulling a creature for this to work!</span>")
+		return FALSE
+	if(!iscarbon(ooze.pulling))
+		to_chat(src, "<span class='warning'>This creature is not advanced enough to be assisted with a cocoon!</span>")
+		return FALSE
```

Comment:
Couldn't they be pulling a non-creature here?
I'd substitute the two checks with just `iscarbon` and adapt the feedback message

Reply:
what? No. Carbons are monkey alien and human.

Reply:
yes, but `!iscarbon(ooze.pulling)` won't just catch `living`, it will also catch any object being pulled as well